pinecone/index_connection_test.go

```diff
@@ -243,6 +244,196 @@ func (ts *IndexConnectionTests) TestListVectors() {
 	assert.NotNil(ts.T(), res)
 }
 
+func TestMarshalingFetchVectorsResponse(t *testing.T) {
+	tests := []struct {
+		name  string
+		input FetchVectorsResponse
+		want  string
+	}{
+		{
+			name: "All fields present",
+			input: FetchVectorsResponse{
+				Vectors: map[string]*Vector{
+					"vec-1": {Id: "vec-1", Values: []float32{0.01, 0.01, 0.01}},
+					"vec-2": {Id: "vec-2", Values: []float32{0.02, 0.02, 0.02}},
+				},
+				Usage: &Usage{ReadUnits: toUInt32(5)},
+			},
+			want: `{"vectors":{"vec-1":{"id":"vec-1","values":[0.01,0.01,0.01]},"vec-2":{"id":"vec-2","values":[0.02,0.02,0.02]}},"usage":{"read_units":5}}`,
+		},
+		{
+			name:  "Fields omitted",
+			input: FetchVectorsResponse{},
+			want:  `{}`,
```

Comment:
Should we account for a case when some fields are present, but others are not (or are empty)?

Reply:
There are cases where we do test for that like in `TestMarshalingDescribeIndexStatsResponse`, but for something like this where all fields have `omitempty` we just test for an empty struct.